src/ComposerRequireChecker/NodeVisitor/UsedSymbolCollector.php

```diff
@@ -52,6 +52,7 @@ public function enterNode(Node $node)
         $this->recordFunctionReturnTypeUsage($node);
         $this->recordConstantFetchUsage($node);
         $this->recordTraitUsage($node);
+        $this->recordProperyTypeUsage($node);
```

Comment:
This is the problematic code path:

```diff
5) /home/runner/work/ComposerRequireChecker/ComposerRequireChecker/src/ComposerRequireChecker/NodeVisitor/UsedSymbolCollector.php:55    [M] MethodCallRemoval

--- Original
+++ New
@@ @@
         $this->recordFunctionReturnTypeUsage($node);
         $this->recordConstantFetchUsage($node);
         $this->recordTraitUsage($node);
-        $this->recordProperyTypeUsage($node);
+        
         return parent::enterNode($node);
     }
     private function recordExtendsUsage(Node $node) : void
```

This path seems to not be tested at all, with current unit tests (integration tests have `@coversNothing`, hence a unit test is needed)

Reply:
Out of interest, why do integration tests have `@coversNothing`?

Reply:
Integration tests "leak" coverage, as in they provide partial coverage for a very wide surface area, but are supposed to be treated as "smoke test" for the whole thing, rather than in-depth testing.

In mutation tests, they lead to extremely slow mutation test runs, and therefore it is quite important to exclude them, favouring instead smaller and more efficient unit tests.